hbase-server/src/test/java/org/apache/hadoop/hbase/master/TestRegionsRecoveryConfigManager.java

```diff
@@ -0,0 +1,147 @@
+    conf.unset("hbase.regions.recovery.store.file.ref.count");
+    conf.unset("hbase.master.regions.recovery.check.interval");
+    StartMiniClusterOption option = StartMiniClusterOption.builder()
+      .masterClass(TestHMaster.class)
+      .numRegionServers(1)
+      .numDataNodes(1).build();
+    HBASE_TESTING_UTILITY.startMiniCluster(option);
+    cluster = HBASE_TESTING_UTILITY.getMiniHBaseCluster();
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    HBASE_TESTING_UTILITY.shutdownMiniCluster();
+  }
+
+  @Test
+  public void testChoreSchedule() throws Exception {
+
+    this.hMaster = cluster.getMaster();
+
+    Stoppable stoppable = new StoppableImplementation();
+    this.regionsRecoveryChore = new RegionsRecoveryChore(stoppable, conf, hMaster);
+
+    this.regionsRecoveryConfigManager = new RegionsRecoveryConfigManager(this.hMaster);
+    // not yet scheduled
+    Assert.assertFalse(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+
+    this.regionsRecoveryConfigManager.onConfigurationChange(conf);
+    // not yet scheduled
+    Assert.assertFalse(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+
+    conf.setInt("hbase.master.regions.recovery.check.interval", 10);
+    this.regionsRecoveryConfigManager.onConfigurationChange(conf);
+    // not yet scheduled - missing config: hbase.regions.recovery.store.file.ref.count
+    Assert.assertFalse(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+
+    conf.setInt("hbase.regions.recovery.store.file.ref.count", 10);
+    this.regionsRecoveryConfigManager.onConfigurationChange(conf);
+    // chore scheduled
+    Assert.assertTrue(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+
+    conf.setInt("hbase.regions.recovery.store.file.ref.count", 20);
+    this.regionsRecoveryConfigManager.onConfigurationChange(conf);
+    // chore re-scheduled
+    Assert.assertTrue(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+
+    conf.setInt("hbase.regions.recovery.store.file.ref.count", 20);
+    this.regionsRecoveryConfigManager.onConfigurationChange(conf);
+    // chore scheduling untouched
+    Assert.assertTrue(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+
+    conf.unset("hbase.regions.recovery.store.file.ref.count");
+    this.regionsRecoveryConfigManager.onConfigurationChange(conf);
+    // chore un-scheduled
+    Assert.assertFalse(hMaster.getChoreService().isChoreScheduled(regionsRecoveryChore));
+  }
+
+  // Make it public so that JVMClusterUtil can access it.
+  public static class TestHMaster extends HMaster {
+    public TestHMaster(Configuration conf) throws IOException, KeeperException {
+      super(conf);
+    }
+  }
+
+  /**
+   * Simple helper class that just keeps track of whether or not its stopped.
+   */
+  private static class StoppableImplementation implements Stoppable {
+
+    private volatile boolean stop = false;
```

Comment:
is this volatile necessary here? thanks

Reply:
since we have just one test method and one thread accessing it, it is not absolutely necessary, just good to have and also, not harmful. If you are fine, we can keep it?

Reply:
+1